src/account/operations/retry.rs

```diff
@@ -0,0 +1,129 @@
+                );
+            }
+
+            let block_id = match transaction.block_id {
+                Some(block_id) => block_id,
+                None => self
+                    .client
+                    .block()
+                    .finish_block(Some(Payload::Transaction(Box::new(transaction.payload.clone()))))
+                    .await?
+                    .id(),
+            };
+
+            // Attachments of the Block to check inclusion state
+            let mut block_ids = vec![block_id];
+            for _ in 0..max_attempts.unwrap_or(DEFAULT_RETRY_UNTIL_INCLUDED_MAX_AMOUNT) {
+                #[cfg(target_family = "wasm")]
+                gloo_timers::future::TimeoutFuture::new(
+                    (interval.unwrap_or(DEFAULT_RETRY_UNTIL_INCLUDED_INTERVAL) * 1000)
+                        .try_into()
+                        .unwrap(),
+                )
+                .await;
+
+                #[cfg(not(target_family = "wasm"))]
+                tokio::time::sleep(std::time::Duration::from_secs(
+                    interval.unwrap_or(DEFAULT_RETRY_UNTIL_INCLUDED_INTERVAL),
+                ))
+                .await;
+
+                // Check inclusion state for each attachment
+                let block_ids_len = block_ids.len();
+                let mut conflicting = false;
+                for (index, block_id_) in block_ids.clone().iter().enumerate() {
+                    let block_metadata = self.client.get_block_metadata(block_id_).await?;
+                    if let Some(inclusion_state) = block_metadata.ledger_inclusion_state {
+                        match inclusion_state {
+                            LedgerInclusionStateDto::Included | LedgerInclusionStateDto::NoTransaction => {
+                                return Ok(*block_id_);
+                            }
+                            // only set it as conflicting here and don't return, because another reattached block could
+                            // have the included transaction
+                            LedgerInclusionStateDto::Conflicting => conflicting = true,
+                        };
+                    }
+                    // Only reattach or promote latest attachment of the block
+                    if index == block_ids_len - 1 {
+                        if block_metadata.should_promote.unwrap_or(false) {
+                            // Safe to unwrap since we iterate over it
+                            self.client.promote_unchecked(block_ids.last().unwrap()).await?;
+                        } else if block_metadata.should_reattach.unwrap_or(false) {
+                            let reattached_block = self
+                                .client
+                                .block()
+                                .finish_block(Some(Payload::Transaction(Box::new(transaction.payload.clone()))))
+                                .await?;
+                            block_ids.push(reattached_block.id());
```

Comment:
Just checking, do you want this to be part of next iterations of the loop line 90 ? Because I'm not sure it's happening as `block_ids.clone().iter().enumerate()` is probably evaluated only once.

Reply:
Yes, why do you think that it's probably evaluated only once?

Reply:
I think that's how `for` works, it creates the iterator only once

Reply:
But there is another loop in L74 `for _ in 0..max_attempts.unwrap_or(DEFAULT_RETRY_UNTIL_INCLUDED_MAX_AMOUNT) {`

Reply:
Yes but my question was if you want these newly added blocks to be part of the current loop or not

Reply:
Current for L90 *

Reply:
Ah no, could be done, but wasn't my plan

Reply:
Ok, was just making sure